### ISSUE TITLE
Double '$' characters in passwords are being truncated

### ISSUE THREAD

**Issue by shieronymus-slingshot** [httpyac/httpyac.github.io#57]
I'm having issues where a password with 2 '$' characters in a row is getting truncated to just 1 -- to get it to work, I have to use 3 '$' characters in a row.

I'm using the VS Code extension on an M1 Mac, but it happened on an x86-64 PC also.

The HTTP request is a standard GET request using Basic Auth, with a password pulled from an httpYac environment variable in the VS Code settings file. So a request header like this:

```Authorization: Basic {{username}}:{{password}}```

Let me know if you need more details. Thank you.

**Comment by AnWeber** [comment]
@shieronymus-slingshot Yes, I'm afraid I need help to recreate it. It currently works correctly for me. I have the following test setup. The password is correctly determined and sent

> auth.http
```
GET https://httpbin.org/basic-auth/{{user}}/{{password}} HTTP/1.1
Authorization: Basic {{user}} {{password}}
```
> settings.json (VSCode Settings)
```json
  "httpyac.environmentVariables": {
    "$shared": {
      "user": "doe",
      "password": "$$doe_secure"
    }
  }
```

I can also imagine a possible problem only in connection with `{{$...}}`. Could you please adapt my example for this reason, so that it doesn't work anymore?

**Comment by shieronymus-slingshot** [comment]
Hmm, after playing with your new test case, I also am unable to replicate the problem. Sorry for the noise!

I think the issue must lie in the web/api setup I'm using.

Thanks for all the work on your tool. 
-Seth

**Comment by shieronymus-slingshot** [comment]
Actually, I don't think it's working... playing with the example some more I see an issue. If you just do the request as above, it does work. But if you deliberately cause it to fail, but using an invalid password in the Authentication line, you can see the issue. 

For example, I have my settings like this:
```
    "httpyac.environmentVariables": {
        "$shared": {
            "user": "user",
            "password": "pa$$word"
        }
    }
```

But if I use the following request (with bad auth line):

```
###
GET https://httpbin.org/basic-auth/{{user}}/{{password}} HTTP/1.1
Authorization: Basic {{user}} bad_pa$$word
```

You get the following output (note the password in the GET line -- it's missing a '$' character):

```
GET https://httpbin.org/basic-auth/user/pa$word
accept-encoding: gzip, deflate, br
accept: */*
authorization: Basic dXNlcjpiYWRfcGEkJHdvcmQ=
user-agent: httpyac

HTTP/1.1 401 UNAUTHORIZED
access-control-allow-credentials: true
access-control-allow-origin: *
connection: close
content-length: 0
date: Fri, 28 Oct 2022 22:36:09 GMT
server: gunicorn/19.9.0
www-authenticate: Basic realm="Fake Realm"
```

Could the variable expansion be swallowing one of the '$' characters?

**Comment by AnWeber** [comment]
@shieronymus-slingshot I really love Javascript/ Typescript, but some Functions are really broken. Why does [replace](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace?retiredLocale=de) remove the `$`. Executing this statement in the dev tools returns `pa$word`

```
"{{password}}".replace("{{password}}", "pa$$word")
```

¯\_(ツ)_/¯

**Comment by AnWeber** [comment]
I published v5.7.6 fixing issue with string.replace
